### ISSUE TITLE
parallel model instead of parallel data

### ISSUE THREAD

**Issue by BaohaoLiao** [pytorch/fairseq#1236]
Hi,

the default multi-gpu training for fairseq is parallel dataset. Now I have two big models, I want to put one model in one GPU, and there are some relations between the output of thesse two model. For example student-teacher architecture, student in one GPU, teacher in another GPU, and compute the loss between the output of these two models. Do you have convenient way to do this?

**Comment by ngoyal2707** [comment]
@BaohaoLiao We currently don't have automated model parallelism in fairseq. I would suggest you start with pytorch's tutorial: https://pytorch.org/tutorials/intermediate/model_parallel_tutorial.html . 
You might have to change a bit of fairseq code but should be simple enough for above use-case.
